### PR TITLE
✨ add id on rum events

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

Add id on rum events to better trace them in the backend

## Changes

- Add ids to views, resources, long tasks, errors, actions

## Testing

Unit, local
---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum-core/src/domain/rumEventsCollection/view/viewCollection.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ function processViewUpdate(view: ViewEvent) {
     date: preferredTimeStamp(view.startClocks),
     type: RumEventType.VIEW,
     view: {
+      id: view.id,
```
Comment (bcaudan) [inline packages/rum-core/src/domain/rumEventsCollection/view/viewCollection.ts:29]: Adding the id here does not seem useful since it will be overridden later
what about not adding it?

Comment (amortemousque) [reply packages/rum-core/src/domain/rumEventsCollection/view/viewCollection.ts:29]: I can't see where it is overriden?

File: packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts [hunk 2]
```diff
@@ -48,6 +48,7 @@ function processRequest(request: RequestCompleteEvent) {
     {
       date: preferredTimeStamp(startClocks),
       resource: {
+        id: generateUUID(),
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts:51]: Since now all resources have an id, you could remove the `resource.id` from the `computeRequestTracingInfo` function below.

File: packages/rum-core/src/domain/rumEventsCollection/action/actionCollection.ts [hunk 3]
```diff
@@ -44,6 +44,7 @@ function processAction(action: AutoAction | CustomAction) {
   const actionEvent = combine(
     {
       action: {
+        id: generateUUID(),
```
Comment (bcaudan) [inline packages/rum-core/src/domain/rumEventsCollection/action/actionCollection.ts:47]: What about generating/adding it only for custom actions?

File: packages/rum-core/src/domain/assembly.spec.ts [hunk 4]
```diff
@@ -298,7 +298,7 @@ describe('rum assembly', () => {
         rawRumEvent: createRawRumEvent(RumEventType.ACTION),
         startTime: 0 as RelativeTime,
       })
-      expect((serverRumEvents[0] as RumActionEvent).action.id).not.toBeDefined()
+      expect((serverRumEvents[0] as RumActionEvent).action.id).toBeDefined()
```
Comment (bcaudan) [inline packages/rum-core/src/domain/assembly.spec.ts:301]: what is the intent of this expectation?

File: packages/rum-core/src/domain/rumEventsCollection/view/viewCollection.spec.ts [hunk 5]
```diff
@@ -39,7 +39,7 @@ describe('viewCollection', () => {
         resourceCount: 10,
         userActionCount: 10,
       },
-      id: 'xxx',
+      id: 'aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee',
```
Comment (bcaudan) [inline packages/rum-core/src/domain/rumEventsCollection/view/viewCollection.spec.ts:42]: what is the purpose of this change?

Comment (amortemousque) [reply packages/rum-core/src/domain/rumEventsCollection/view/viewCollection.spec.ts:42]: The view id has to respect a certain format compliant with rum-events-format

Comment (bcaudan) [reply packages/rum-core/src/domain/rumEventsCollection/view/viewCollection.spec.ts:42]: 🤦‍♂️ indeed

File: packages/rum-core/src/domain/rumEventsCollection/action/actionCollection.ts (hunk 3 above)
Comment (amortemousque) [reply packages/rum-core/src/domain/rumEventsCollection/action/actionCollection.ts:47]: I'm probably missing a point. I don't really see why we only have to target custom actions?

File: packages/rum-core/src/domain/assembly.spec.ts (hunk 4 above)
Comment (amortemousque) [reply packages/rum-core/src/domain/assembly.spec.ts:301]: Shouldn't the id be required for all actions?

File: packages/rum-core/src/domain/rumEventsCollection/action/actionCollection.ts (hunk 3 above)
Comment (bcaudan) [reply packages/rum-core/src/domain/rumEventsCollection/action/actionCollection.ts:47]: action id for auto action is generated earlier in the code and retrieved in `autoActionProperties`.
With the current code, for an auto action, an id is generated and overwritten immediately by the id from `autoActionProperties`.

IMO, we could make the intent clearer by having something like:
```ts
const actionTypeProperties = isAutoAction(action)
    ? {
        action: {
          error: {
            count: action.counts.errorCount,
          },
          id: action.id,
          loading_time: toServerDuration(action.duration),
          long_task: {
            count: action.counts.longTaskCount,
          },
          resource: {
            count: action.counts.resourceCount,
          },
        },
      }
    : {
        action: {
          id: generateUUID()
        }
      }
```

File: packages/rum-core/src/domain/assembly.spec.ts (hunk 4 above)
Comment (bcaudan) [reply packages/rum-core/src/domain/assembly.spec.ts:301]: yes, but it is not the goal of this test.
In the context of this test, the idea is to check by event type is the action context is added to the event or not, meaning testing [this function](https://github.com/DataDog/browser-sdk/blob/abccf2c/packages/rum-core/src/domain/assembly.ts#L127)

Comment (amortemousque) [reply packages/rum-core/src/domain/assembly.spec.ts]: okay done ;)
